src/library/scala/collection/immutable/List.scala

```diff
@@ -332,6 +333,67 @@ sealed abstract class List[+A]
     acc
   }
 
+  // Copy/Paste overrides to avoid interface calls inside loops.
```

Comment:
Why did we not see the effect of this in our micro-benchmarks? Is it because they are all testing monomorphic cases?

Reply:
Perhaps. JMH has different warmup modes to help test this theory.

```
   -wm <mode>                  Warmup mode for warming up selected benchmarks.
                               Warmup modes are: INDI = Warmup each benchmark individually,
                               then measure it. BULK = Warmup all benchmarks first,
                               then do all the measurements. BULK_INDI = Warmup
                               all benchmarks first, then re-warmup each benchmark
                               individually, then measure it. (default: INDI)

   -wmb <regexp+>              Warmup benchmarks to include in the run in addition
                               to already selected by the primary filters. Harness
                               will not measure these benchmarks, but only use them
                               for the warmup.
```